conformance/utils/roundtripper/roundtripper.go

```diff
@@ -153,3 +179,14 @@ func formatDump(data []byte, prefix string) string {
 	data = startLineRegex.ReplaceAllLiteral(data, []byte(prefix))
 	return string(data)
 }
+
+func IsRedirect(statusCode int) bool {
+	// Gateway allows only 301, and 302
```

Comment:
Even though we only allow 301 and 302, I'd make this logic more future-proof to include the full set of redirect codes. There's a chance we'll add more in the future, but even if not, it will be more helpful to show a more useful error message than if we didn't capture the redirect info.

Reply:
Can you elaborate? my thought was to put it in a function so we will have a one central place for adding more codes in the future. I think now we want to fail conformance if someone returns something other than 301/302 because we do not support it, correct?

Reply:
Let's say an implementation is actually returning a 307 or 308. With the current code, we wouldn't actually populate the rest of the redirect info for the rest of the test logic, so it would just look like _everything_ failed instead of just having the wrong status code set. I think the `IsRedirect` function is primarily a check on if we should populate the new redirect struct. The roundtripper itself should not care about what is conformant or not, just in making sure it populates as much information as possible so the tests can determine if the response was conformant.

Reply:
Thats a good point, however we do check for wrong status code https://github.com/kubernetes-sigs/gateway-api/blob/c1c4952732d0acbcf294ce93c511ff86a6380c4d/conformance/utils/http/http.go#L194-L196

I agree the roundtripper should not care about conformance, would you prefer to construct the redirect struct always? Whats the value you think it'll give us? it wont change the behavior of the above lines - we will have a failure with a wrong status code still.

Reply:
I think we should always construct the redirect struct for any 3xx status code. Although the test will still fail, it will likely provide more useful information to the person debugging the test. Instead of simply knowing the code is wrong, they'll also be able to see if the location they are redirecting to has any issues.

Reply:
Makes sense, I have added all the codes but if you prefer I will do something like this instead let me know
```golang
func IsRedirect(statusCode int) bool {
	switch {
	case statusCode >= 300 && statusCode < 400:
		return true
	}
	return false
}
```